pygmt/src/pygmtlogo.py

```diff
@@ -0,0 +1,289 @@
+        ``True`` for a hexagon shape.
+    wordmark : bool
+        Add the wordmark "PyGMT". ``True`` adds the wordmark [Default] and ``False``
+        to skipes it.
+    orientation : str
+        Orientation of the wordmark. Use ``"horizontal"`` to place the wordmark at
+        the right side [Default] or ``"vertical"`` to place it below the visual.
+    """
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    # Start of subfunction
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    def create_logo(  # noqa: PLR0915
+        blackwhite=blackwhite,
+        darkmode=darkmode,
+        hexshape=hexshape,
+        wordmark=wordmark,
+        orientation=orientation,
+    ):
+        # -----------------------------------------------------------------------------
+        # Helpful definitions
+        # -----------------------------------------------------------------------------
+        size = 4
+        region = [-size, size] * 2
+
+        xy_yellow_1 = 2.65
+        xy_yellow_2 = 1.3
+
+        no_line = "cyan@100"
+        no_fill = f"+g{no_line}"
+
+        # Rotation around z (vertical) axis placed in the center
+        # Has to be applied to each plotting command, up on second call set to True
+        # Do NOT rotated initial basemap
+        angle_rot = 30  # degrees
+        perspective = f"{angle_rot}+w0/0"
+
+        # -----------------------------------------------------------------------------
+        # Define colors
+        # -----------------------------------------------------------------------------
+        color_dark = "gray20"
+        color_light = "white"
+
+        # visual
+        color_blue = "48/105/152"  # Python blue
+        color_yellow = "255/212/59"  # Python yellow
+        color_red = "238/86/52"  # GMT red
+        if blackwhite:
+            color_blue = color_yellow = color_red = color_light
+            if not darkmode:
+                color_blue = color_yellow = color_red = color_dark
+
+        # background and wordmark
+        color_bg = color_dark
+        color_py = color_yellow
+        color_gmt = color_light
+        if not darkmode:
+            color_bg = color_light
+            color_py = color_blue
+            color_gmt = color_dark
+
+        # -----------------------------------------------------------------------------
+        # Define shape
+        # -----------------------------------------------------------------------------
+        symbol = "c"  # circle
+        diameter = 7.5
+        diameter_add = 0.5
+        if hexshape:
+            symbol = "h"  # hexagon
+            diameter = 8.6
+            diameter_add = 0.6
+
+        # -----------------------------------------------------------------------------
+        # Define wordmark
+        # -----------------------------------------------------------------------------
+        font = "AvantGarde-Book"
```

Comment:
The font *Space Grotesk* by Florian Karsten is not available in GMT. Currently the font *AvantGarde-Book* is used.